src/api/topics/utils.py

```diff
@@ -19,3 +21,28 @@ def parse_order_by(value: str) -> dict[str, str] | None:  # noqa: CFQ004  # pyli
     if parameter not in Topic.SORTING_FIELDS or order not in Topic.SORTING_ORDER:
         return abort(400, "provided value is not allowed")
     return {"field": parameter, "order": order}
+
+
+def parse_author_id(value: str) -> int:
+    """Parse author_id query parameter."""
+    if not value:
+        return abort(400, "empty argument value is not allowed")
+    return int(value)
+
+
+def parse_datetime_args(value: str) -> datetime:
+    """Parse dates as strings and return datetime objects."""
+    if not value:
+        return abort(400, "empty argument value is not allowed")
+    return datetime.strptime(value, "%Y-%m-%d %H:%M:%S.%f")
```

Comment:
I think we need to also check that value is a parsable datetime string. Because for now user can send and arbitrary string as datetime query parameter and it will cause an internal server error.